### PR TITLE
(webdriver): find good known version when Chromedriver can't be installed with found `buildId`

### PR DESCRIPTION
**Pull request by christian-bromann**
## Proposed changes

We have observed in CI that a Chrome browser is found on the machine, e.g. `v115.0.5790.171` but a driver with the same `buildId` doesn't exist. In this case let's make the driver manager a bit more versatile and have it detect the next known good version.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

n/a

### CODE REVIEW

**Comment by christian-bromann** [comment]
@harsha509 yes, this is the same case. When WebdriverIO detects a Chrome browser on the machine it doesn't seem to be guaranteed that there is a Chromedriver with the same `buildId`. For this case this patch will find a last good known version for it to execute.

File: packages/webdriver/src/driver/index.ts [hunk 1]
```diff
@@ -85,17 +86,33 @@ export async function startWebDriver (options: Options.WebDriver) {
...
+                 */
+                log.warn(`Couldn't download Chromedriver v${buildId}: ${err.message}, trying to find known good version...`)
+                const majorVersion = buildId.split('.')[0]
+                const knownGoodVersions: any = await got('https://googlechromelabs.github.io/chrome-for-testing/known-good-versions.json').json()
+                const knownGoodVersion = knownGoodVersions.versions.filter(({ version }: { version: string }) => version.startsWith(majorVersion)).pop()
+                if (!knownGoodVersion) {
+                    throw new Error(`Couldn't find known good version for Chromedriver v${majorVersion}`)
+                }
+                loggedBuildId = knownGoodVersion.version
+                return install({ ...chromedriverInstallOpts, buildId: loggedBuildId })
```
Comment (HannaTarasevich) [inline packages/webdriver/src/driver/index.ts:115]: Can we have a case when known good version has already been installed and can be used from cache?

**Review by HannaTarasevich — COMMENTED** [review]
Based on this PR, I'm closing this one https://github.com/webdriverio/webdriverio/pull/10899

File: packages/webdriver/src/driver/index.ts (hunk 1 above)
Comment (HannaTarasevich) [reply packages/webdriver/src/driver/index.ts:115]: So this way, console won't contain message about the version of Chromedriver in use, right?
Like we have for else flow in row 118

**Pull request merged**